File: PerfTest1/PerfTest1-Xamarin/PerfTest1-ClassicXamarin.Android/AzureList.cs
using System.Collections.Generic;
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;

namespace PerfTest1Xamarin
{
    public class AzureList : Fragment
    {

        public AzureList()
        {
            // Required empty public constructor
        }

        public override void OnStart()
        {
            base.OnStart();
            Button btnClear = (Button) this.Activity.FindViewById(Resource.Id.btnClear);
            Button btnFetch = (Button) this.Activity.FindViewById(Resource.Id.btnFetch);

            btnClear.SetOnClickListener((View.IOnClickListener) this.Activity);
            btnFetch.SetOnClickListener((View.IOnClickListener) this.Activity);

            ListView lstRegistrations = (ListView) this.Activity.FindViewById(Resource.Id.lstRegistrations);
            lstRegistrations.Adapter = new RegistrationAdapter(this.Activity, new List<Registration>());
        }

        public override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState)
        {
            // Inflate the layout for this fragment
            return inflater.Inflate(Resource.Layout.fragment_azure_list, container, false);
        }

        public override void OnDetach()
        {
            base.OnDetach();
            Button btnClear = (Button) this.Activity.FindViewById(Resource.Id.btnClear);
            Button btnFetch = (Button) this.Activity.FindViewById(Resource.Id.btnFetch);

            if (btnClear != null)
            {
                btnClear.SetOnClickListener(null);
            }
            if (btnFetch != null)
            {
                btnFetch.SetOnClickListener(null);
            }
        }
    }
}

File: PerfTest1/PerfTest1-Java/app/src/main/java/com/vandammeford/kevinf/perftest1_java/AzureList.java
package com.vandammeford.kevinf.perftest1_java;

import android.os.Bundle;
import android.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ListView;

import java.util.ArrayList;


public class AzureList extends Fragment {

    public AzureList() {
        // Required empty public constructor
    }

    @Override
    public void onStart() {
        super.onStart();
        Button btnClear = (Button)this.getActivity().findViewById(R.id.btnClear);
        Button btnFetch = (Button)this.getActivity().findViewById(R.id.btnFetch);

        btnClear.setOnClickListener((View.OnClickListener)this.getActivity());
        btnFetch.setOnClickListener((View.OnClickListener) this.getActivity());

        ListView lstRegistrations = (ListView)this.getActivity().findViewById(R.id.lstRegistrations);
        lstRegistrations.setAdapter(new RegistrationAdapter(this.getActivity(), new ArrayList<Registration>()));
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_azure_list, container, false);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Button btnClear = (Button)this.getActivity().findViewById(R.id.btnClear);
        Button btnFetch = (Button)this.getActivity().findViewById(R.id.btnFetch);

        if (btnClear != null) {
            btnClear.setOnClickListener(null);
        }
        if (btnFetch != null) {
            btnFetch.setOnClickListener(null);
        }
    }
}
